services/proxy/pkg/middleware/mfa.go

```diff
@@ -0,0 +1,73 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	"github.com/owncloud/ocis/v2/ocis-pkg/mfa"
+	"github.com/owncloud/ocis/v2/ocis-pkg/oidc"
+	"github.com/owncloud/ocis/v2/services/proxy/pkg/config"
+)
+
+// MultiFactor returns a middleware that checks requests for mfa
+func MultiFactor(cfg config.MFAConfig, opts ...Option) func(next http.Handler) http.Handler {
+	options := newOptions(opts...)
+	logger := options.Logger
+
+	return func(next http.Handler) http.Handler {
+		return &MultiFactorAuthentication{
+			next:           next,
+			logger:         logger,
+			enabled:        cfg.Enabled,
+			authLevelNames: cfg.AuthLevelNames,
+		}
+	}
+}
+
+// MultiFactorAuthentication is a authenticator that checks for mfa on specific paths
+type MultiFactorAuthentication struct {
+	next           http.Handler
+	logger         log.Logger
+	enabled        bool
+	authLevelNames []string
+}
+
+// ServeHTTP adds the mfa header if the request contains a valid mfa token
+func (m MultiFactorAuthentication) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	defer m.next.ServeHTTP(w, req)
+
+	if !m.enabled {
+		mfa.SetHeader(req, true)
+		return
+	}
+
+	// overwrite the mfa header to avoid passing on wrong information
+	mfa.SetHeader(req, false)
```

Comment:
This seems counter-intuitive. If the middleware isn't enabled? why are we setting the header to true?
If it has to be like this, I think we need to include a comment explaining because I'm sure someone will switch the values later.

Reply:
We are setting the header to true so the system does not need to care if mfa is enabled or disabled. It can assume mfa is always on. I'll add a comment explaining this.

If you mean the other `SetHeader(req, false)`. We are doing this to overwrite the header if it is already set.

Reply:
Added a comment to explain behaviour